python/lsst/ts/scheduler/scheduler_csc.py

```diff
@@ -1579,3 +1676,277 @@ def callback_script_info(self, data):
                 del self.script_info[key]
                 if len(self.script_info) < self.parameters.max_scripts:
                     break
+
+    async def register_observation(self, target: DriverTarget) -> None:
+        """Register observation.
+
+        Parameters
+        ----------
+        observation : Observation
+            Observation to be registered.
+        """
+        self.driver.register_observation(target=target)
+        if hasattr(self, "evt_observation"):
+            await self.evt_observation.set_write(
+                **dataclasses.asdict(target.get_observation())
+            )
+
+    async def _handle_driver_configure_scheduler(
+        self, config: typing.Any
+    ) -> SurveyTopology:
+        """Handle configuring the scheduler asynchronously.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Configuration, as described by ``schema/Scheduler.yaml``
+
+        Returns
+        -------
+        survey_topology: `SurveyTopology`
+            Survey topology
+        """
+
+        configure_scheduler = functools.partial(
+            self.driver.configure_scheduler, config=config
+        )
+
+        return await asyncio.get_running_loop().run_in_executor(
+            None, configure_scheduler
+        )
+
+    async def _handle_load_snapshot(self, uri: str) -> None:
+        """Handler loading a scheduler snapshot asynchronously.
+
+        Parameters
+        ----------
+        uri : str
+            Uri with the address of the snapshot.
+        """
+        loop = asyncio.get_running_loop()
+
+        try:
+            retrieve = functools.partial(
+                urllib.request.urlretrieve, url=uri, filename=""
+            )
+            dest, _ = await loop.run_in_executor(None, retrieve)
+        except urllib.request.URLError:
+            raise RuntimeError(
+                f"Could not retrieve {uri}. Make sure it is a valid and accessible URI."
```

Comment:
It looks like you're mixing `uri`, `Uri` and `URI` in all the Scheduler code. To be frank, I don't know if `URI` is preferred or not so just leave it all as is.